pkg/controller/chaosengine/chaosengine_controller.go

```diff
@@ -753,137 +735,167 @@ func (r *ReconcileChaosEngine) removeChaosRunner(engine *chaosTypes.EngineInfo,
 	return nil
 }
 
-func (r *ReconcileChaosEngine) checkRunnerPodForCompletion(engine *chaosTypes.EngineInfo, reqLogger logr.Logger) error {
-	isCompleted := r.checkRunnerPodCompleted(engine)
-	if isCompleted {
-		err := r.updateStatus(engine, litmuschaosv1alpha1.EngineStatusCompleted)
-		if err != nil {
-			return err
-		}
-	}
-	return nil
-}
-
-func checkEngineStatusForComplete(engine *chaosTypes.EngineInfo) bool {
-	return engine.Instance.Status.EngineStatus == litmuschaosv1alpha1.EngineStatusCompleted
-
-}
-
+// reconcileForComplete reconciles for graceful completion of Chaos Engine
 func (r *ReconcileChaosEngine) reconcileForComplete(request reconcile.Request) (reconcile.Result, error) {
-	deletetimeStamp := engine.Instance.ObjectMeta.GetDeletionTimestamp()
-	if deletetimeStamp != nil {
-		return r.reconcileForDelete(request)
-	}
-	_, err := r.removeDefaultChaosResources(request)
+
+	_, err := r.gracefullyRemoveDefaultChaosResources(request)
 	if err != nil {
 		r.recorder.Eventf(engine.Instance, corev1.EventTypeWarning, "ChaosResourcesOperationFailed", "Unable to delete chaos resources")
 		return reconcile.Result{}, err
 	}
-	err = r.updateState(engine, litmuschaosv1alpha1.EngineStateStop)
+	err = r.updateEngineState(engine, litmuschaosv1alpha1.EngineStateStop)
 	if err != nil {
 		r.recorder.Eventf(engine.Instance, corev1.EventTypeWarning, "ChaosResourcesOperationFailed", "Unable to update chaosengine")
 		return reconcile.Result{}, err
 	}
 	return reconcile.Result{}, nil
 }
 
-func (r *ReconcileChaosEngine) initEngineState(engine *chaosTypes.EngineInfo) error {
-	if engine.Instance.Spec.EngineState == "" {
-		err := r.updateState(engine, litmuschaosv1alpha1.EngineStateActive)
-		return err
+// reconcileForRestart reconciles for restart of Chaos Engine
+func (r *ReconcileChaosEngine) reconcileForRestart(request reconcile.Request) (reconcile.Result, error) {
+	err := r.forceRemoveChaosResources(engine, request)
```

Comment:
Do we need to remove the chaos resources in case of restart?? as we have already deleted the resources when we stop that earlier??

Reply:
Ok, so if we are restarting the engine, it means that everything related to engine has to cleaned up. This is the case for `stopped` or `completed` engine, for stopped all resources are deleted, but for `completed` state, it depends on `jobCleanUpPolicy`.